### ISSUE TITLE
no-undef rule not working in .eslintrc in yaml format

### ISSUE THREAD

**Issue by PhiLhoSoft** [eslint/eslint#4928]
I have found a very strange bug...

I use ESLint v1.10.3 on Windows 7.

In a project, I made a `.eslintrc.yaml` files with all the rules of ESLint, some activated for my specific needs. Of course, I have the `no-undef: 2` rule!
It works fine.
So I copied the settings and pasted them in a `.eslintrc` file in another project: it was in Json format, I just change the content but didn't want to rename the file to reduce noise.
Then I adjusted some settings to adapt to the other project (different style), but kept the `no-undef` one.
When editing the file, I noticed I get all kinds of warnings, according to my settings, but I could just create as many global variables as I wanted!

After lot of investigation, I eliminated all combinations of settings... And finally found out it was the extension of the file that was the culprit! In other words, after renaming the file to add the `.yaml` extension, I got the warning...

```bash
> ls .es*
.eslintignore  .eslintrc
> eslint src/main/js/
(nothing reported)
> mv .eslintrc .eslintrc.yaml
> eslint src/main/js/

C:\PhiLhoSoft\Web\Playground\src\main\js\main.js
  2:1  error  "a" is not defined  no-undef

✖ 1 problem (1 error, 0 warnings)
```
I thought the extension was just a "decoration", and to help editors to trigger the right syntax highlighting... And why only this setting? (of those I use, I didn't check everything...)

So I report this, even if I have a workaround...

**Comment by gyandeeps** [comment]
Can you run eslint with `--debug` flag and share the output for both cases. this will tell us what all config files are getting used.

**Comment by PhiLhoSoft** [comment]
No, I don't have multiple config files in the same dir (I have one in the test sub-dir, only).
Coming back to report that after renaming the file, it revealed other errors: max-params, max-len, and max-statements.
So the issue isn't specific to no-undef.

Errors I saw reported while in no-prefix format: linebreak-style, indent, quote, semi, space-before-function-paren, and probably some others.

**Comment by PhiLhoSoft** [comment]
```
> mv .eslintrc.yaml .eslintrc

> eslint src/main/js/

> eslint --debug src/main/js/
  eslint:cli Running on files +0ms
  eslint:glob-util Creating list of files to process. +57ms
  eslint:cli-engine Processing C:\PhiLhoSoft\Web\Playground\src\main\js\App.js +20ms
  eslint:cli-engine Linting C:\PhiLhoSoft\Web\Playground\src\main\js\App.js +4ms
  eslint:config Constructing config for C:\PhiLhoSoft\Web\Playground\src\main\js\App.js +1ms
  eslint:config Using .eslintrc and package.json files +0ms
  eslint:config Loading C:\PhiLhoSoft\Web\Playground\.eslintrc +5ms
  eslint:config-file Loading config file: C:\PhiLhoSoft\Web\Playground\.eslintrc +2ms
  eslint:config-ops Apply environment settings to config +162ms
  eslint:config-ops Creating config for environment browser +0ms
  eslint:config-ops Creating config for environment es6 +1ms
  eslint:config-ops Creating config for environment jquery +0ms
  eslint:config Using C:\PhiLhoSoft\Web\Playground\.eslintrc +2ms
  eslint:config Loading C:\PhiLhoSoft\Web\Playground\package.json +0ms
  eslint:config-file Loading package.json config file: C:\PhiLhoSoft\Web\Playground\package.json +0ms
  eslint:config Merging command line environment settings +2ms
  eslint:cli-engine Processing C:\PhiLhoSoft\Web\Playground\src\main\js\Cow.js +64ms
  eslint:cli-engine Linting C:\PhiLhoSoft\Web\Playground\src\main\js\Cow.js +1ms
  eslint:config Constructing config for C:\PhiLhoSoft\Web\Playground\src\main\js\Cow.js +1ms
  eslint:config Using config from cache +0ms
  eslint:cli-engine Processing C:\PhiLhoSoft\Web\Playground\src\main\js\main.js +32ms
  eslint:cli-engine Linting C:\PhiLhoSoft\Web\Playground\src\main\js\main.js +1ms
  eslint:config Constructing config for C:\PhiLhoSoft\Web\Playground\src\main\js\main.js +0ms
  eslint:config Using config from cache +0ms
  eslint:cli-engine Processing C:\PhiLhoSoft\Web\Playground\src\main\js\store\reduxRedux.js +10ms
  eslint:cli-engine Linting C:\PhiLhoSoft\Web\Playground\src\main\js\store\reduxRedux.js +1ms
  eslint:config Constructing config for C:\PhiLhoSoft\Web\Playground\src\main\js\store\reduxRedux.js +0ms
  eslint:config Using .eslintrc and package.json files +0ms
  eslint:config Loading C:\PhiLhoSoft\Web\Playground\.eslintrc +2ms
  eslint:config-file Loading config file: C:\PhiLhoSoft\Web\Playground\.eslintrc +0ms
  eslint:config-ops Apply environment settings to config +10ms
  eslint:config-ops Creating config for environment browser +22ms
  eslint:config-ops Creating config for environment es6 +2ms
  eslint:config-ops Creating config for environment jquery +0ms
  eslint:config Using C:\PhiLhoSoft\Web\Playground\.eslintrc +1ms
  eslint:config Loading C:\PhiLhoSoft\Web\Playground\package.json +0ms
  eslint:config-file Loading package.json config file: C:\PhiLhoSoft\Web\Playground\package.json +1ms
  eslint:config Merging command line environment settings +0ms
  eslint:cli-engine Processing C:\PhiLhoSoft\Web\Playground\src\main\js\utility\ComponentWithLoadingLifecycle.js +40ms
  eslint:cli-engine Linting C:\PhiLhoSoft\Web\Playground\src\main\js\utility\ComponentWithLoadingLifecycle.js +1ms
  eslint:config Constructing config for C:\PhiLhoSoft\Web\Playground\src\main\js\utility\ComponentWithLoadingLifecycle.js +1ms
  eslint:config Using .eslintrc and package.json files +0ms
  eslint:config Loading C:\PhiLhoSoft\Web\Playground\.eslintrc +1ms
  eslint:config-file Loading config file: C:\PhiLhoSoft\Web\Playground\.eslintrc +1ms
  eslint:config-ops Apply environment settings to config +9ms
  eslint:config-ops Creating config for environment browser +1ms
  eslint:config-ops Creating config for environment es6 +1ms
  eslint:config-ops Creating config for environment jquery +0ms
  eslint:config Using C:\PhiLhoSoft\Web\Playground\.eslintrc +1ms
  eslint:config Loading C:\PhiLhoSoft\Web\Playground\package.json +0ms
  eslint:config-file Loading package.json config file: C:\PhiLhoSoft\Web\Playground\package.json +1ms
  eslint:config Merging command line environment settings +0ms
  eslint:cli-engine Processing C:\PhiLhoSoft\Web\Playground\src\main\js\utility\Helpers.js +28ms
  eslint:cli-engine Linting C:\PhiLhoSoft\Web\Playground\src\main\js\utility\Helpers.js +1ms
  eslint:config Constructing config for C:\PhiLhoSoft\Web\Playground\src\main\js\utility\Helpers.js +0ms
  eslint:config Using config from cache +1ms
  eslint:cli-engine Processing C:\PhiLhoSoft\Web\Playground\src\main\js\utility\MathUtilities.js +13ms
  eslint:cli-engine Linting C:\PhiLhoSoft\Web\Playground\src\main\js\utility\MathUtilities.js +0ms
  eslint:config Constructing config for C:\PhiLhoSoft\Web\Playground\src\main\js\utility\MathUtilities.js +1ms
  eslint:config Using config from cache +0ms
  eslint:cli-engine Linting complete in: 476ms +25ms

> mv .eslintrc .eslintrc.yaml

> eslint src/main/js/

C:\PhiLhoSoft\Web\Playground\src\main\js\main.js
  2:1  error  "a" is not defined  no-undef

? 1 problem (1 error, 0 warnings)


> eslint --debug src/main/js/
  eslint:cli Running on files +0ms
  eslint:glob-util Creating list of files to process. +56ms
  eslint:cli-engine Processing C:\PhiLhoSoft\Web\Playground\src\main\js\App.js +21ms
  eslint:cli-engine Linting C:\PhiLhoSoft\Web\Playground\src\main\js\App.js +2ms
  eslint:config Constructing config for C:\PhiLhoSoft\Web\Playground\src\main\js\App.js +1ms
  eslint:config Using .eslintrc and package.json files +1ms
  eslint:config Loading C:\PhiLhoSoft\Web\Playground\.eslintrc.yaml +4ms
  eslint:config-file Loading YAML config file: C:\PhiLhoSoft\Web\Playground\.eslintrc.yaml +1ms
  eslint:config-ops Apply environment settings to config +314ms
  eslint:config-ops Creating config for environment browser +1ms
  eslint:config-ops Creating config for environment es6 +0ms
  eslint:config-ops Creating config for environment jquery +1ms
  eslint:config Using C:\PhiLhoSoft\Web\Playground\.eslintrc.yaml +1ms
  eslint:config Loading C:\PhiLhoSoft\Web\Playground\package.json +1ms
  eslint:config-file Loading package.json config file: C:\PhiLhoSoft\Web\Playground\package.json +0ms
  eslint:config Merging command line environment settings +2ms
  eslint:cli-engine Processing C:\PhiLhoSoft\Web\Playground\src\main\js\Cow.js +57ms
  eslint:cli-engine Linting C:\PhiLhoSoft\Web\Playground\src\main\js\Cow.js +1ms
  eslint:config Constructing config for C:\PhiLhoSoft\Web\Playground\src\main\js\Cow.js +0ms
  eslint:config Using config from cache +1ms
  eslint:cli-engine Processing C:\PhiLhoSoft\Web\Playground\src\main\js\main.js +35ms
  eslint:cli-engine Linting C:\PhiLhoSoft\Web\Playground\src\main\js\main.js +1ms
  eslint:config Constructing config for C:\PhiLhoSoft\Web\Playground\src\main\js\main.js +0ms
  eslint:config Using config from cache +0ms
  eslint:cli-engine Processing C:\PhiLhoSoft\Web\Playground\src\main\js\store\reduxRedux.js +12ms
  eslint:cli-engine Linting C:\PhiLhoSoft\Web\Playground\src\main\js\store\reduxRedux.js +1ms
  eslint:config Constructing config for C:\PhiLhoSoft\Web\Playground\src\main\js\store\reduxRedux.js +1ms
  eslint:config Using .eslintrc and package.json files +0ms
  eslint:config Loading C:\PhiLhoSoft\Web\Playground\.eslintrc.yaml +1ms
  eslint:config-file Loading YAML config file: C:\PhiLhoSoft\Web\Playground\.eslintrc.yaml +0ms
  eslint:config-ops Apply environment settings to config +10ms
  eslint:config-ops Creating config for environment browser +1ms
  eslint:config-ops Creating config for environment es6 +1ms
  eslint:config-ops Creating config for environment jquery +0ms
  eslint:config Using C:\PhiLhoSoft\Web\Playground\.eslintrc.yaml +1ms
  eslint:config Loading C:\PhiLhoSoft\Web\Playground\package.json +1ms
  eslint:config-file Loading package.json config file: C:\PhiLhoSoft\Web\Playground\package.json +0ms
  eslint:config Merging command line environment settings +1ms
  eslint:cli-engine Processing C:\PhiLhoSoft\Web\Playground\src\main\js\utility\ComponentWithLoadingLifecycle.js +45ms
  eslint:cli-engine Linting C:\PhiLhoSoft\Web\Playground\src\main\js\utility\ComponentWithLoadingLifecycle.js +0ms
  eslint:config Constructing config for C:\PhiLhoSoft\Web\Playground\src\main\js\utility\ComponentWithLoadingLifecycle.js +1ms
  eslint:config Using .eslintrc and package.json files +0ms
  eslint:config Loading C:\PhiLhoSoft\Web\Playground\.eslintrc.yaml +1ms
  eslint:config-file Loading YAML config file: C:\PhiLhoSoft\Web\Playground\.eslintrc.yaml +1ms
  eslint:config-ops Apply environment settings to config +4ms
  eslint:config-ops Creating config for environment browser +1ms
  eslint:config-ops Creating config for environment es6 +1ms
  eslint:config-ops Creating config for environment jquery +0ms
  eslint:config Using C:\PhiLhoSoft\Web\Playground\.eslintrc.yaml +1ms
  eslint:config Loading C:\PhiLhoSoft\Web\Playground\package.json +0ms
  eslint:config-file Loading package.json config file: C:\PhiLhoSoft\Web\Playground\package.json +1ms
  eslint:config Merging command line environment settings +1ms
  eslint:cli-engine Processing C:\PhiLhoSoft\Web\Playground\src\main\js\utility\Helpers.js +30ms
  eslint:cli-engine Linting C:\PhiLhoSoft\Web\Playground\src\main\js\utility\Helpers.js +1ms
  eslint:config Constructing config for C:\PhiLhoSoft\Web\Playground\src\main\js\utility\Helpers.js +0ms
  eslint:config Using config from cache +0ms
  eslint:cli-engine Processing C:\PhiLhoSoft\Web\Playground\src\main\js\utility\MathUtilities.js +23ms
  eslint:cli-engine Linting C:\PhiLhoSoft\Web\Playground\src\main\js\utility\MathUtilities.js +1ms
  eslint:config Constructing config for C:\PhiLhoSoft\Web\Playground\src\main\js\utility\MathUtilities.js +1ms
  eslint:config Using config from cache +0ms
  eslint:cli-engine Linting complete in: 614ms +24ms

C:\PhiLhoSoft\Web\Playground\src\main\js\main.js
  2:1  error  "a" is not defined  no-undef

? 1 problem (1 error, 0 warnings)
```

**Comment by gyandeeps** [comment]
based on the diff (https://www.diffchecker.com/ssyf3jbu), its picking up the file correctly.

**Comment by ilyavolodin** [comment]
Extensions are not just decorators for syntax highlighting. We support config files in JSON, YAML and JS format. Extension will let us know how to parser the file to be able to load it's content. So I think this is behaving as expected.

**Comment by PhiLhoSoft** [comment]
> Deprecated - use .eslintrc, which can be either JSON or YAML.

OK, it is deprecated, but it is supposed to be still supported. And it is picking a number of settings, just not all of them.

**Comment by gyandeeps** [comment]
Oh yeah, lol i completly missed it.
@PhiLhoSoft When you change your file from .eslintrc to .eslinrc.yaml then you have to change the format of data inside the file also.

**Comment by gyandeeps** [comment]
Can you share the content of that file?

**Comment by PhiLhoSoft** [comment]
I wonder if I have a syntax error in my yaml file. I cannot reproduce the issue with a very simple config.
Mmm, no, http://www.yamllint.com/ tells me it is OK.

> When you change your file from .eslintrc to .eslinrc.yaml then you have to change the format of data inside the file also.

I don't understand. It is Yaml in both cases.

Here is the file, with .txt extension to please GitHub, just remove that.
[.eslintrc.yaml.txt](https://github.com/eslint/eslint/files/87541/default.eslintrc.yaml.txt)

**Comment by gyandeeps** [comment]
I actually tried the same thing on my machine and i was not able to reproduce the error.

**Comment by PhiLhoSoft** [comment]
OK, I found the culprit!
The line:

  
```
spaced-comment: 1             # require or disallow a space immediately following the # or /* in a comment
```
Apparently, in a `.eslintrc` file, the `/*` makes the remainder of the file to be commented out!
Not in a `.eslintrc.yaml` one...

**Comment by gyandeeps** [comment]
I see. Tricky. :stuck_out_tongue_closed_eyes:
